### PR TITLE
KEYCLOAK-14158 Polish My Resource page

### PR DESCRIPTION
**Pull request by edewit**
fixed a lot of small issues each one is a seperate commit for easy review

### CODE REVIEW

**Comment by stianst** [comment]
@edewit Please check the https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md - and please also make sure the GitHub PR description explains what it is about otherwise everyone has to look at the PR and JIRA to figure out what this is about

**Comment by edewit** [comment]
@stianst you mean i did not have a single commit? I did that on purpose as it makes for an easier review. The issue: `KEYCLOAK-14158` contains a large number of small issues we could have split those all up into separate issues alternatively. When the review is done I will rebase and squash.

**Comment by stianst** [comment]
@edewit if the plan is to squash to a single commit before merge that works

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx [hunk 1]
```diff
@@ -127,7 +127,9 @@ export class PermissionRequest extends React.Component<PermissionRequestProps, P
                                                 {shareRequest.scopes.map((scope, j) => <Chip key={j} isReadOnly>{scope}</Chip>)}
                                             </DataListCell>,
                                             <DataListCell key={`actions${i}`}>
-                                                <Button onClick={() => this.handleApprove(shareRequest.username, shareRequest.scopes)}>
+                                                <Button
+                                                    style={{marginRight: '10px'}}
```
Comment (ssilvert) [inline themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx:131]: Can we find a PatternFly way to deal with this?  Granted, I haven't run this to see what it is about, but this sort of thing is always a red flag.

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ export class PermissionRequest extends React.Component<PermissionRequestProps, P
         const permissions: Permission[] = permissionsRequest.data || [];
 
         // Erik - I had to add the exclamation point.  Can we be sure that the 'find' will not return undefined?
```
Comment (ssilvert) [inline themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx:61]: Do we know the answer to my question here?

**Comment by ssilvert** [comment]
Does the work you did here on Alerts fully fix https://issues.redhat.com/browse/KEYCLOAK-12005?

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx (hunk 2 above)
Comment (edewit) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx:61]: yep it's not always there I fixed that in #7067

**Comment by edewit** [comment]
> Does the work you did here on Alerts fully fix https://issues.redhat.com/browse/KEYCLOAK-12005?

Yes it does

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx (hunk 1 above)
Comment (edewit) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/PermissionRequest.tsx]: changed to use Stack

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/ResourcesTable.tsx [hunk 3]
```diff
@@ -83,6 +87,16 @@ export class ResourcesTable extends AbstractResourcesTable<CollapsibleResourcesT
...
 
     public render(): React.ReactNode {
+        if (this.props.resources.data.length === 0) {
+            return (
+                <EmptyState variant={EmptyStateVariant.full}>
+                    <EmptyStateIcon icon={RepositoryIcon}/>
+                    <Title headingLevel="h5" size="lg">
+                        <Msg msgKey="notHaveAnyResource"/>
+                    </Title>
+                </EmptyState>
```
[the remark this answers is not in the document]
Comment (edewit) [reply themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/ResourcesTable.tsx]: done

**Comment by edewit** [comment]
UI test: https://keycloak-jenkins.com/job/keycloak-account2-matrix/903/

**Comment by ssilvert** [comment]
The UI test run is good enough.  The two failing tests are not related to this PR, so I removed the "Run UI Tests" label.

**Review by pedroigor — DISMISSED** [review]

**Review by ssilvert — APPROVED** [review]

**Comment by vmuzikar** [comment]
@ssilvert Btw, I was just about to review it (see my email from Friday). ;)

**Review by vmuzikar — COMMENTED** [review]
@edewit Thank you for the redesign. It looks much better and it's much more easy and intuitive to use now. I like it, nice work!

I've, however, found a few bugs.
* When you unshare a resource, the change doesn't take effect in the UI imediately. It still says that "Resource is share with ...".
* It's not possible to unshare a resource for a selected user (well, technically it is, but you need to remove all scopes from given user which is not very intuitive). When you unshare, it's unshared for all users. If the button is really supposed to unshare all, it should explicitly say so and there should be a warning modal dialog.
* When a user is not found, a generic "Not Found" alert is displayed **after** the Share dialog is closed. There are two problems with that. 1) "Not Found" is not explanatory enough (should be perhaps "User XY was not found". 2) When you share a resource with multiple users at once, only a success alert is displayed. User is not informed at all that some other user was not found and resource is not shared. We probably check for user existence when adding them using the "Add" button in the dialog.
* The Permission requests dialog doesn't seem to work at all! Accept/Deny doesn't do anything.

Also, there are still some valid issues stated in the original JIRA.
* There are still some missing capitals. "C" in "close" in permissions request dialog. "P" in "permissions requested" in the same dialog. "P" in "permissions" in Shared with me table.
* Permission request dialog doesn't show Requestor.

**Comment by ssilvert** [comment]
> @ssilvert Btw, I was just about to review it (see my email from Friday). ;)

@vmuzikar  So sorry.  I screwed up.  We will address the issues above in https://issues.redhat.com/browse/KEYCLOAK-14414

Thanks for the review.

**Comment by vmuzikar** [comment]
@ssilvert No worries, thanks!

**Comment by edewit** [comment]
I like that you review and find new issues, that you didn't find before.
The reason I didn't fix "Permission request dialog doesn't show Requestor."
is because I couldn't reproduce it. I think you are looking at the data
from the UI test where the requests only have a boolean. BTW next time
instead of writing an email put a comment on the PR, that way all the
information about merging is central.

**Comment by vmuzikar** [comment]
> The reason I didn't fix "Permission request dialog doesn't show Requestor." is because I couldn't reproduce it. I think you are looking at the data from the UI test where the requests only have a boolean.
Ok, thanks. I'll double check it – that might be it.

> BTW next time instead of writing an email put a comment on the PR, that way all the information about merging is central.
I was replying to Stan's email (you were on CC) where he was asking for a review. But ok, next time I'll both reply to an email and copy-paste it as a PR comment as well to make super explicit. ;)

**Pull request merged**